src/sema/ast.rs

```diff
@@ -268,13 +272,15 @@ pub struct Function {
     /// The resolved body (if any)
     pub body: Vec<Statement>,
     pub symtable: Symtable,
-    // What events are emitted by the body of this function
+    /// What events are emitted by the body of this function
     pub emits_events: Vec<usize>,
-    // For overloaded functions this is the mangled (unique) name.
+    /// For overloaded functions this is the mangled (unique) name.
     pub mangled_name: String,
+    /// Which contracts should we use the mangled name in?
+    pub use_mangle_name_on: HashSet<usize>,
```

Comment:
The name does not include `contracts` which is essential. How about `mangled_name_contracts`.